src/CrossReferences.jl

```diff
@@ -127,7 +127,7 @@ function docsxref(link::Markdown.Link, code, meta, page, doc)
             return
         end
     end
-    local mod = get(meta, :CurrentModule, current_module())
+    local mod = get(meta, :CurrentModule, @__MODULE__())
```

Comment:
Unfortunately I don't think we can do a `current_module()` to `@__MODULE__` replacement -- they're not equivalent. Right now building e.g. Documenter's own docs with this branch will crash because the modules don't get resolved correctly.

```julia
julia> module Foo
       f() = (@__MODULE__, current_module())
       end

julia> Foo.f()
WARNING: current_module() is deprecated, use `@__MODULE__` instead.
Stacktrace:
 [1] depwarn(::String, ::Symbol) at ./deprecated.jl:68
 [2] current_module() at ./deprecated.jl:1359
 [3] f() at ./REPL[11]:2
 [4] eval(::Module, ::Expr) at ./repl/REPL.jl:3
 [5] eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./repl/REPL.jl:69
 [6] macro expansion at ./repl/REPL.jl:100 [inlined]
 [7] (::getfield(Base.REPL, Symbol("##1#2")){Base.REPL.REPLBackend})() at ./event.jl:96
while loading no file, in expression starting on line 0
(Foo, Main)
```